ketchup/main.py

```diff
@@ -1,121 +1,231 @@
+                try:
+                    episodes = int(input(
+                        "How many episodes are you interested in watching?: "))
+                except ValueError:
+                    print("Sorry, please only enter a value greater than 0.")
+                    continue
+                if episodes <= 0:
+                    print("Sorry, please only enter a value greater than 0.")
+                    continue
+                else:
+                    break
 
-    end_date_prompt = (
-        "What date would you like to have completed this series?(ex: 15 Aug 2050): "
-    )
-    end_date = get_date(end_date_prompt, require_future=True)
-
-    answer_today = input("Will you start watching today? [Y/n]: ") or "y"
-    if answer_today.lower() == "y":
-        start_date = dt.datetime.today()
-    else:
-        start_date_prompt = "What day will you start?(15 Aug 2023): "
-        start_date = get_date(start_date_prompt)
-
-    per_day_limit = (
-        input("What is the limit of episodes you will watch per day? [0+] ") or 0
-    )
-
-    # The following seems as though it could be better...
-    # We want to set the per day limit as an integer, or None...
-    try:
-        per_day_limit = int(per_day_limit)
-        if per_day_limit == 0:
-            per_day_limit = None
-    except ValueError:
-        per_day_limit = None
-
-    # Time to math!
-    watch_schedule = wsb.calculate_watch_schedule(
-        episodes, end_date, start_date, per_day_limit
-    )
-    print(f"Wow, that's only {watch_schedule['days']} days between today and then !!!")
-
-    if watch_schedule["is_possible"]:
-        print(
-            f"You will have to watch {watch_schedule['base_eps_per_day']} episodes per day "
-            "to complete the show on time!"
-        )
-        if watch_schedule["extra_episodes"]:
-            print(
-                f"You will have {watch_schedule['extra_episodes']} extra,"
-                "you can add an additional watch day or tack them on to one of your other days."
+            end_date_prompt = (
+                "What date would you like to have completed this series? "
+                "(Example: 15 Aug 2050): "
             )
-    else:
-        print("I'm sorry, but that limit and end date make it impossible!")
-        print("Don't worry, consider these options instead:")
-        print(
-            f"Option 1: You can watch {watch_schedule['base_eps_per_day']} episodes per day to catch up"
-        )
-        if watch_schedule["extra_episodes"]:
-            print(
-                f"You will have {watch_schedule['extra_episodes']} extra,"
-                "you can add an additional watch day or tack them on to one of your other days."
+
+            answer_today = input(
+                "Will you start watching today? [Y/N]: ") or "y"
+
+            if answer_today.lower() == "y":
+                start_date = dt.datetime.today()
+                episode_duration_mins = 24
```

Comment:
with the calculate function you added above, we expressed:
1) Episode Duration should be settable by customer
2) calculate function should be in utils.

Therefore; for this specific line, we need to either inherit the duration value here or... ?